src/sentry/api/endpoints/organization_dashboards.py

```diff
@@ -48,15 +48,17 @@ def has_object_permission(self, request: Request, view, obj):
             return super().has_object_permission(request, view, obj)
 
         if isinstance(obj, Dashboard):
-            # 1. Dashboard contains certain projects
-            if obj.projects.exists():
-                return request.access.has_projects_access(obj.projects.all())
-
-            # 2. Dashboard covers all projects or all my projects
-
-            # allow when Open Membership
-            if obj.organization.flags.allow_joinleave:
-                return True
+            if not features.has(
+                "organizations:dashboards-edit-access", obj.organization, actor=request.user
+            ):
+                # 1. Dashboard contains certain projects
+                if obj.projects.exists():
+                    return request.access.has_projects_access(obj.projects.all())
+
+                # 2. Dashboard covers all projects or all my projects
+                # allow when Open Membership
+                if obj.organization.flags.allow_joinleave:
+                    return True
```

Comment:
The reason we don't want to allow this case is because we want to move towards more explicit dashboard permissions via teams. Users should need to assign themselves to a team to get edit perms, regardless of open membership or project assignments. This gives us clearer control over permissions, as team membership is the real factor, not just project coverage. Plus, all data is already queryable, so project-based restrictions are just a proxy for what we really want—team-based permissions. Also, this permissions class is used to only restrict edit and delete access, so anyone can still view any dashboard.